### PR TITLE
docs(proposals): wrote out the proposal journey in mermaid

### LINKED ISSUE #11639: Document next iteration of proposal process
We have an existing proposal process that attempts to provide a way for consumers of Carbon to provide suggestions for changes to the Carbon Design System. Most often, this takes place in the form of feature requests for existing components or new components altogether.

The current setup has the following structure:

- An issue is created by the contributor
- The issue is triaged into the "proposal queue" through the "proposal: open" label
- During backlog cleaning, the team will look at proposals and will make one of the following decisions:
  - Needs more research: the issue is unclear or we're not sure if this is something we would like to pursue as a team
  - Accepted: the team agrees with the overall feature request or component and believes it should be in Carbon
  - Not pursuing: the team does not believe this feature request or component is something that reasonably can exist in Carbon at this time

This setup has several drawbacks that lead to a negative experience for a contributor, including:

- Lack of clarity or awareness of this process, a contributor might ask "what does it mean to be a proposal or to have it be accepted?"
- Lack of clarity around what is expected from the contributor if a proposal is stalled
- Lack of clarity around the timeframe for which an accepted proposal will be delivered

The purpose of this issue is to:

- [x] Collect feedback from existing contributors to better understand the current limitations and expectations around this process
- [x] Synthesize this feedback into pain points
- [x] Deliver a to-be scenario for this proposal process that we could try out as a team next sprint
- [x] Determine what success metrics, if any, we should evaluate changes to this process by

### PR DESCRIPTION
**Pull request by sstrubberg**
Closes #11639 

#### Feedback

- Are there any parts of the journey that are confusing?
- Do the definitions help?
- I chose Green60 for the moments where we are labeling things in Github. Red60 for when the issue is closed. Purple70 with the stroke-dash-array to illustrate the "arduous" process to harden a spec. Mermaid [basic syntax](https://mermaid-js.github.io/mermaid/#/flowchart?id=basic-support-for-fontawesome) if you have other ideas. I can't use emojis (or FontAwesome as the Mermaid docs suggest for whatever reason).

#### Testing

image

### CODE REVIEW

File: docs/guides/support.md [hunk 1]
```diff
@@ -218,7 +218,68 @@ missing info.
...
+
+Guiding principless -
+https://carbondesignsystem.com/all-about-carbon/what-is-carbon/#our-guiding-principles
+
+DSAG - The Design System Adoption Guild is an expansive group of designers and
+developers all aligned on the mission of delivering amazing experiences in the
+IBM Products organization. You can learn more about the guild
+[here](https://pages.github.ibm.com/cdai-design/pal/contributing/contribution-framework/intent).
+
+PAL - Pattern Access Library. These are the local systems around IBM that extend
```
Comment (elycheea) [inline docs/guides/support.md]: I think this should be Pattern and Asset Library. 🙃

Comment (sstrubberg) [reply docs/guides/support.md]: Blargh.. you're totally right!

**Review by tw15egan — APPROVED** [review]
LGTM, very cool chart didn't know you could do that in markdown 🤯

**Review by abbeyhrt — APPROVED** [review]

**Comment by ljcarot** [comment]
Looks good @sstrubberg.  Let's merge this and try it out.  Can always improve or iterate from this starting point!

**Review by laurenmrice — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]
🧜‍♀️

**Comment by aagonzales** [comment]
@sstrubberg  In general, I approve but one question I think people might ask regarding the first bubble. 

> Does it align with the maintainer team's guiding principles for building a design system at IBM?

What are the guiding principles? 

---
I also wonder if the last `Yes` should lead to "Closed as complete" and to that point if the other "Close" should be "Closed as not persuing," like how github defines the closes. Just to add clarity 

image

**Comment by sstrubberg** [comment]
> @sstrubberg In general, I approve but one question I think people might ask regarding the first bubble.
> 
> > Does it align with the maintainer team's guiding principles for building a design system at IBM?
> 
> What are the guiding principles? UPDATE: Oh wait! I see them now they are down below. Can you add links in bubbles?
> 
> I also wonder if the last `Yes` should lead to "Closed as complete" and to that point if the other "Close" should be "Closed as not persuing," like how github defines the closes. Just to add clarity
> 
> image

I believe I can add some links to the bubbles. I'll try in a new PR. Separately, I'll mock up a version where it's "Closed as complete" vs "Closed as not pursuing" to see how it feels.

**Pull request merged**